spec/factories/stripe/stripe_plans.rb

```diff
@@ -6,12 +6,13 @@
 
     transient do 
       sequence(:id) {|i| "test_str_plan#{i}"}
+      product { association :stripe_product }
```

Comment:
`StripePlan`s in Stripe 5 must have a product associated with this. All of the changes handle that.